examples/Health/HealthDiagnostics.cpp

```diff
@@ -0,0 +1,138 @@
+#include "HealthDiagnostics.h"
+
+String HealthDiagnostics::getChipId() {
+#if defined(ESP32)
+  return String((uint32_t)ESP.getEfuseMac(), HEX);
+#elif defined(ESP8266)
+  return String(ESP.getChipId(), HEX);
+#elif defined(ARDUINO_ARCH_RP2040)
+  return String(rp2040.getChipID(), HEX);
+#endif
+}
+
+void HealthDiagnostics::addHeapInfo(JsonObject& doc) {
+#if defined(ESP32)
+  doc["freeHeap"] = ESP.getFreeHeap();
+  doc["totalHeap"] = ESP.getHeapSize();
+  doc["minFreeHeap"] = ESP.getMinFreeHeap();
+  doc["maxAllocHeap"] = ESP.getMaxAllocHeap();
+
+  multi_heap_info_t heap_info;
+  heap_caps_get_info(&heap_info, MALLOC_CAP_INTERNAL);
+
+  JsonObject internalHeap = doc["internalHeap"].to<JsonObject>();
+  internalHeap["totalFreeBytes"] = heap_info.total_free_bytes;
+  internalHeap["totalAllocatedBytes"] = heap_info.total_allocated_bytes;
+  internalHeap["largestFreeBlock"] = heap_info.largest_free_block;
+  internalHeap["minimumFreeBytes"] = heap_info.minimum_free_bytes;
+  internalHeap["allocatedBlocks"] = heap_info.allocated_blocks;
+  internalHeap["freeBlocks"] = heap_info.free_blocks;
+  internalHeap["totalBlocks"] = heap_info.total_blocks;
+
+  #ifdef CONFIG_SPIRAM_SUPPORT
```

Comment:
`CONFIG_SPIRAM_SUPPORT` is always defined, except your're on an IDF project where this has been undefined in menuconfig.